File: app/src/main/java/vn/loitp/app/activity/customviews/layout/scrollview2d/LHorizontalScrollView.java
package vn.loitp.app.activity.customviews.layout.scrollview2d;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.HorizontalScrollView;

import androidx.annotation.NonNull;

public class LHorizontalScrollView extends HorizontalScrollView {

    private ScrollListener mScrollListener;

    public LHorizontalScrollView(Context context) {
        super(context);
    }

    public LHorizontalScrollView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public LHorizontalScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    public void setOnScrollListener(@NonNull final ScrollListener scrollListener) {
        this.mScrollListener = scrollListener;
    }

    @Override
    protected void onScrollChanged(int l, int t, int oldl, int oldt) {
        super.onScrollChanged(l, t, oldl, oldt);
        if (mScrollListener != null) {
            mScrollListener.onScrollChange(this, l, t, oldl, oldt);
        }
    }

    public interface ScrollListener {
        void onScrollChange(View view, int scrollX, int scrollY, int oldScrollX, int oldScrollY);
    }
}

File: core/src/main/java/com/loitpcore/views/scrollView/LHorizontalScrollView.kt
package com.loitpcore.views.scrollView

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.HorizontalScrollView

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
class LHorizontalScrollView : HorizontalScrollView {

    @Suppress("unused")
    interface ScrollListener {
        fun onScrollChange(view: View, scrollX: Int, scrollY: Int, oldScrollX: Int, oldScrollY: Int)
    }

    private var mScrollListener: LScrollView.ScrollListener? = null

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    fun setOnScrollListener(scrollListener: LScrollView.ScrollListener) {
        this.mScrollListener = scrollListener
    }

    override fun onScrollChanged(l: Int, t: Int, oldl: Int, oldt: Int) {
        super.onScrollChanged(l, t, oldl, oldt)

        mScrollListener?.onScrollChange(this, l, t, oldl, oldt)
    }
}
